### ISSUE TITLE
Program install error! 

### ISSUE THREAD

**Issue by Ghostboy-287** [Ekultek/Zeus-Scanner#2]
Hello i got this error while trying to use zeus script :/ 
```
~$ sudo ./zeus.py -d inutl:php?id=

[03:35:34 INFO] using default search engine...
[03:35:34 INFO] starting dork scan with query '.php?id='...
[03:35:34 INFO] attempting to gather query URL...
[03:35:34 ERROR] search.pyc failed to gather the URL from search engine, caught exception 'cmd=['Xvfb', '-help']
OSError=[Errno 2] No such file or directory
Program install error!' exception has been logged to current log file...
Traceback (most recent call last):
  File "/home/username/Pentest/Zeus-Scanner/var/google_search/search.py", line 170, in parse_search_results
    query_url = get_urls(query, url, verbose=verbose, user_agent=user_agent, proxy=proxy_string)
  File "/home/username/Pentest/Zeus-Scanner/var/google_search/search.py", line 36, in get_urls
    ff_display = Display(visible=0, size=(800, 600))
  File "/usr/local/lib/python2.7/dist-packages/pyvirtualdisplay/display.py", line 34, in __init__
    self._obj = self.display_class(
  File "/usr/local/lib/python2.7/dist-packages/pyvirtualdisplay/display.py", line 52, in display_class
    cls.check_installed()
  File "/usr/local/lib/python2.7/dist-packages/pyvirtualdisplay/xvfb.py", line 38, in check_installed
    ubuntu_package=PACKAGE).check_installed()
  File "/usr/local/lib/python2.7/dist-packages/easyprocess/__init__.py", line 180, in check_installed
    raise EasyProcessCheckInstalledError(self)
EasyProcessCheckInstalledError: cmd=['Xvfb', '-help']
OSError=[Errno 2] No such file or directory
Program install error! 
```
**Additional info:**
- Os: Elementary Os (based on ubuntu)
- sqlmap and nmap installed and updated 
- all requirements have been installed

**Comment by Ekultek** [comment]
Hi, thanks for using `zeus` give me a few minutes to look over this

**Comment by Ekultek** [comment]
Alright so this seems to be an issue with `xvfb` not being installed on your system, try doing this: `sudo apt-get install xvfb` and from there run `python zeus.py -d <DORK> --verbose` and post the output for me, thank you.

**Comment by Ekultek** [comment]
for some info on the app see here https://linux.die.net/man/1/xvfb

**Comment by Ekultek** [comment]
Also, I'd advise you to be sure that sqlmap and nmap are in your environment PATH to save searching time.

**Comment by Ch0nDa** [comment]
I think the issue was me using a mac OSX terminal and not ubuntu, I will retry in Linux (Kali first, then UBUNTU) to see if it works for me them.  I am noticing already that when i run pip intstall -r requirements.txt there is an installation procedure for Linux, whereas there is none for mac.

**Comment by Ekultek** [comment]
Yes, I haven't gotten around to implementing other operating systems yet. Try it with Linux and see what happens, keep me updated.

**Comment by Ch0nDa** [comment]
Here is some more output

parallels@ubuntu:~/bin/Zeus-Scanner$ sudo python zeus.py -d hello --verbose
[23:48:19 DEBUG] checking if the application has been run before...
[23:48:19 DEBUG] already ran, skipping...

    __          __________                             __   
   / /          \____    /____  __ __  ______          \ \  
  / /    ______   /     // __ \|  |  \/  ___/  ______   \ \ 
  \ \   /_____/  /     /\  ___/|  |  /\___ \  /_____/   / / 
   \_\          /_______ \___  >____//____  >          /_/  
                       \/   \/           \/  v1.0.15.9cd3(revision)
	https://github.com/ekultek/zeus-scanner.git
		Advanced Dork Searching...


[*] starting up at 23:48:19..


[23:48:19 DEBUG] running with options '{'runInVerbose': True, 'dorkToUse': 'hello'}'...
[23:48:19 INFO] log file being saved to '/home/parallels/bin/Zeus-Scanner/log/zeus-log-6.log'...
[23:48:19 DEBUG] using default search engine (Google)...
[23:48:19 INFO] using default search engine...
[23:48:19 INFO] starting dork scan with query 'hello'...
[23:48:19 DEBUG] checking for user-agent and proxy configuration...
[23:48:19 INFO] attempting to gather query URL...
[23:48:19 DEBUG] setting up the virtual display to hide the browser...
[23:48:20 INFO] firefox browser display will be hidden while it performs the query...
[23:48:20 WARNING] your web browser will be automated in order for Zeus to successfully bypass captchas and API calls. this is done in order to grab the URL from the search and parse the results. please give selenium time to finish it's task...
[23:48:20 DEBUG] running selenium-webdriver and launching browser...
[23:48:20 DEBUG] adjusting selenium-webdriver user-agent to 'Zeus-Scanner(v1.0.15.9cd3)::Python->v2.7'...
[23:48:21 ERROR] search.pyc failed to gather the URL from search engine, caught exception 'Message: Unable to find a matching set of capabilities' exception has been logged to current log file...
Traceback (most recent call last):
  File "/home/parallels/bin/Zeus-Scanner/var/google_search/search.py", line 170, in parse_search_results
    query_url = get_urls(query, url, verbose=verbose, user_agent=user_agent, proxy=proxy_string)
  File "/home/parallels/bin/Zeus-Scanner/var/google_search/search.py", line 77, in get_urls
    browser = webdriver.Firefox(profile, proxy=proxy_to_use)
  File "/home/parallels/.local/lib/python2.7/site-packages/selenium/webdriver/firefox/webdriver.py", line 154, in __init__
    keep_alive=True)
  File "/home/parallels/.local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 140, in __init__
    self.start_session(desired_capabilities, browser_profile)
  File "/home/parallels/.local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 229, in start_session
    response = self.execute(Command.NEW_SESSION, parameters)
  File "/home/parallels/.local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 297, in execute
    self.error_handler.check_response(response)
  File "/home/parallels/.local/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 194, in check_response
    raise exception_class(message, screen, stacktrace)
WebDriverException: Message: Unable to find a matching set of capabilities



[*] shutting down at 23:48:21..

**Comment by Ch0nDa** [comment]
how exactly do I bring sqlmap and nmap into the ENV PATH??  I'm missing a piece here since im such a newb

**Comment by lianfeng30** [comment]
kali
[*] starting up at 16:33:35..


[16:33:35 INFO] log file being saved to '/root/Zeus-Scanner/log/zeus-log-4.log'...
[16:33:35 INFO] using default search engine...
[16:33:35 CRITICAL] failed to provide a mandatory argument, you will be redirected to the help menu...

**Comment by Ghostboy-287** [comment]
@Ekultek thank you
installing xvfb using `sudo apt-get install xvfb` command resolved my problem, 
( it is preferable to add it in [requirements](https://github.com/Ekultek/Zeus-Scanner#requirements)

**Comment by Ekultek** [comment]
@Ch0nDa you need to update Firefox. Put sqlmap and nap in /usr/bin

@Ghostboy-287 most of the time xvfb is preinstalled on Linux systems, I can create a check for it though

**Comment by Ekultek** [comment]
@lianfeng30 https://github.com/Ekultek/Zeus-Scanner/wiki

**Comment by Ekultek** [comment]
So I'm going to create a check for Xvfb, look for it in the next push,

**Comment by Ekultek** [comment]
patched via 6875605
